toolkit/tools/pkg/imagecustomizerlib/customizeverity.go

```diff
@@ -0,0 +1,222 @@
+		}
+
+		lines := strings.Split(string(input), "\n")
+		for i, line := range lines {
+			if strings.HasPrefix(line, "mariner_initrd=") {
+				lines[i] = "mariner_initrd=" + newInitramfs
+			}
+		}
+		output := strings.Join(lines, "\n")
+		return ioutil.WriteFile(cfgPath, []byte(output), 0644)
+	})
+
+	return nil
+}
+
+func updateGrubConfig(resolvedVerityDevice string, resolvedHashDevice string, salt string, rootHash string, verityCorruptionResponse string) error {
+	const cmdlineTemplate = "rd.systemd.verity=1 roothash=%s systemd.verity_root_data=%s systemd.verity_root_hash=%s systemd.verity_root_options=%s,salt=%s"
+	newArgs := fmt.Sprintf(cmdlineTemplate, rootHash, resolvedVerityDevice, resolvedHashDevice, verityCorruptionResponse, salt)
+	grubConfigPath := "/mnt/boot_partition/grub2/grub.cfg"
+
+	content, err := ioutil.ReadFile(grubConfigPath)
+	if err != nil {
+		return fmt.Errorf("failed to read grub config: %v", err)
+	}
+
+	// Split the content into lines for processing
+	lines := strings.Split(string(content), "\n")
+	var updatedLines []string
+
+	for _, line := range lines {
+		trimmedLine := strings.TrimSpace(line)
+		if strings.HasPrefix(trimmedLine, "linux ") {
+			// Append new arguments to the line that starts with "linux"
+			line += " " + newArgs
+		}
+		if strings.HasPrefix(trimmedLine, "set rootdevice=PARTUUID=") {
+			// Replace the root device line with the new root device. TODO: add supported type 'user'
+			line = "set rootdevice=/dev/mapper/root"
+		}
+		updatedLines = append(updatedLines, line)
+	}
+
+	// Write the updated content back to grub.cfg
+	err = ioutil.WriteFile(grubConfigPath, []byte(strings.Join(updatedLines, "\n")), 0644)
+	if err != nil {
+		return fmt.Errorf("failed to write updated grub config: %v", err)
+	}
+
+	return nil
+}
+
+// findFreeNBDDevice finds the first available NBD device.
+func findFreeNBDDevice() (string, error) {
+	files, err := filepath.Glob("/sys/class/block/nbd*")
+	if err != nil {
+		return "", err
+	}
+
+	for _, file := range files {
+		size, err := ioutil.ReadFile(filepath.Join(file, "size"))
+		if err != nil {
+			continue
+		}
+
+		if strings.TrimSpace(string(size)) == "0" {
+			return "/dev/" + filepath.Base(file), nil
+		}
+	}
+
+	return "", fmt.Errorf("no free nbd devices available")
+}
+
+// convertToDevicePath takes an NBD device path and a system config device path,
+// and converts it to the corresponding NBD partition path.
+func convertToNbdDevicePath(nbdDevice, systemDevice string) (string, error) {
+	partitionRegex := regexp.MustCompile(`[0-9]+$`)
+	partitionNumber := partitionRegex.FindString(systemDevice)
+
+	if partitionNumber == "" {
+		return "", fmt.Errorf("failed to extract partition number from %s", systemDevice)
+	}
+
+	return fmt.Sprintf("%sp%s", nbdDevice, partitionNumber), nil
+}
+
+// findDeviceByUUIDOrLabel attempts to resolve a PARTUUID, PARTLABEL, UUID, or LABEL to a device file.
+func findDeviceByUUIDOrLabel(uuidOrLabel string) (string, error) {
```

Comment:
I don't think the config should include UUIDs or PARTUUIDs since they are different for different images and aren't human readable. I think you only need to support the partition name.